collector/exporters/mockdestinationexporter/exporter.go

```diff
@@ -0,0 +1,62 @@
+package mockdestinationexporter
+
+import (
+	"context"
+	"errors"
+	"math/rand/v2"
+	"time"
+
+	"go.opentelemetry.io/collector/exporter"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type MockDestinationExporter struct {
+	config *Config
+	logger *zap.Logger
+}
+
+func NewMockDestinationExporter(config *Config,
+	params exporter.Settings) (*MockDestinationExporter, error) {
+
+	if config == nil {
+		return nil, errors.New("mock destination exporter config is nil")
+	}
+
+	logger := params.Logger
+
+	mockDestinationExporter := &MockDestinationExporter{
+		config: config,
+		logger: logger,
+	}
+	return mockDestinationExporter, nil
+}
+
+func (e *MockDestinationExporter) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (e *MockDestinationExporter) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	return e.mockExport(ctx)
+}
+
+func (e *MockDestinationExporter) ConsumeMetrics(ctx context.Context, metrics pmetric.Metrics) error {
+	return e.mockExport(ctx)
+}
+
+func (e *MockDestinationExporter) ConsumeLogs(ctx context.Context, logs plog.Logs) error {
+	return e.mockExport(ctx)
+}
+
+func (e *MockDestinationExporter) mockExport(context.Context) error {
+	// not taking care of ctx cancel and shutdown as this is a dummy exporter and not used in production
+	<-time.After(e.config.ResponseDuration)
```

Comment:
nit: I think time.Sleep is better as time.After create goroutine every time it is called so this has the potential to blow up CPU according to the number of parallel invocation to this function